src/vis/bar/utils.ts

```diff
@@ -108,32 +109,31 @@ export function createBarTraces(
         legendPlots: [],
         rows,
         cols,
-        errorMessage: 'To create a Bar Chart, please select at least 1 categorical column.',
+        errorMessage: I18nextManager.getInstance().i18n.t('tdp:core.vis.barError'),
     };
 }
 
-function setPlotsWithGroupsAndMultiples(columns, config, plots, scales, plotCounter): number {
+async function setPlotsWithGroupsAndMultiples(columns: VisColumn[], catCols: VisCategoricalColumn[], config: IBarConfig, plots: PlotlyData[], scales: Scales, plotCounter: number): Promise<number> {
 
-    const catCols: CategoricalColumn[] = columns.filter((c) => config.catColumnsSelected.filter((d) => c.info.id === d.id).length > 0 && c.type === EColumnTypes.CATEGORICAL) as CategoricalColumn[];
+    const catColValues = await resolveColumnValues(catCols);
     const vertFlag = config.direction === EBarDirection.VERTICAL;
     const normalizedFlag = config.display === EBarDisplayType.NORMALIZED;
-    const catCurr = catCols[0];
-    const currGroupColumn = getCol(columns, config.group) as CategoricalColumn;
-    const currMultiplesColumn = getCol(columns, config.multiples) as CategoricalColumn;
-
+    const catCurr = catColValues[0];
+    const currGroupColumn = await resolveSingleColumn(getCol(columns, config.group));
+    const currMultiplesColumn = await resolveSingleColumn(getCol(columns, config.multiples));
 
-    const uniqueGroupVals = [...new Set(currGroupColumn.values.map((v) => v.val))];
-    const uniqueMultiplesVals = [...new Set(currMultiplesColumn.values.map((v) => v.val))];
+    const uniqueGroupVals = [...new Set(currGroupColumn.resolvedValues.map((v) => v.val))];
+    const uniqueMultiplesVals = [...new Set(currMultiplesColumn.resolvedValues.map((v) => v.val))];
 
-    const uniqueColVals = [...new Set(catCurr.values.map((v) => v.val))];
+    const uniqueColVals = [...new Set(catCurr.resolvedValues.map((v) => v.val))];
 
     uniqueMultiplesVals.forEach((uniqueMultiples) => {
         uniqueGroupVals.forEach((uniqueGroup) => {
 
             const groupedLength = uniqueColVals.map((v) => {
-                const allObjs = catCurr.values.filter((c) => c.val === v).map((c) => c.id);
-                const allGroupObjs = currGroupColumn.values.filter((c) => c.val === uniqueGroup).map((c) => c.id);
-                const allMultiplesObjs = currMultiplesColumn.values.filter((c) => c.val === uniqueMultiples).map((c) => c.id);
+                const allObjs = (catCurr.resolvedValues as {id: number, val:string}[]).filter((c) => c.val === v).map((c) => c.id);
+                const allGroupObjs = (currGroupColumn.resolvedValues as {id: number, val:string}[]).filter((c) => c.val === uniqueGroup).map((c) => c.id);
+                const allMultiplesObjs = (currMultiplesColumn.resolvedValues as {id: number, val:string}[]).filter((c) => c.val === uniqueMultiples).map((c) => c.id);
```

Comment:
Do you have an existing interface somewhere that you could reuse instead of writing `{id: number, val:string}[]` multiple times?

Reply:
created one in interfaces.ts and using it here